testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java

```diff
@@ -0,0 +1,274 @@
+        public void tearDown(final ManagementClient managementClient, final String containerId) throws Exception {
+             removeDss();
+        }
+
+        public void removeDss() {
+            try {
+                remove(dsAddress);
+            } catch (Throwable e) {
+            	log.warn(e.getMessage());
+            }
+            try {
+                remove(dsXaAddress);
+            } catch (Throwable e) {
+            	log.warn(e.getMessage());
+            }
+        }
+
+        /**
+         * Creates data source and return its node address
+         *
+         * @param xa - should be data source XA?
+         * @param jndiName of data source
+         * @return ModelNode - address of data source node
+         * @throws Exception
+         */
+        private ModelNode createDataSource(boolean xa, String jndiName) throws Exception {
+            ModelNode address = new ModelNode();
+            address.add(SUBSYSTEM, "datasources");
+            address.add((xa ? "xa-" : "") + "data-source", jndiName);
+            address.protect();
+
+            ModelNode operation = new ModelNode();
+            operation.get(OP).set(ADD);
+            operation.get(OP_ADDR).set(address);
+            operation.get("jndi-name").set(jndiName);
+            operation.get("driver-name").set("h2");
+            operation.get("enabled").set("false");
+            if (!xa)
+                operation.get("connection-url").set("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1");
+            operation.get("prepared-statements-cache-size").set(3);
+            operation.get("user-name").set("sa");
+            operation.get("password").set("sa");
+            
+            
+            operation.get("connection-listener-class").set(CONNECTION_LISTENER_CLASS_IMPL);
+            executeOperation(operation);
+                   
+            if (xa) {
+                final ModelNode xaDatasourcePropertiesAddress = address.clone();
+                xaDatasourcePropertiesAddress.add("xa-datasource-properties", "URL");
+                xaDatasourcePropertiesAddress.protect();
+                final ModelNode xaDatasourcePropertyOperation = new ModelNode();
+                xaDatasourcePropertyOperation.get(OP).set("add");
+                xaDatasourcePropertyOperation.get(OP_ADDR).set(xaDatasourcePropertiesAddress);
+                xaDatasourcePropertyOperation.get("value").set("jdbc:h2:mem:test");
+
+                executeOperation(xaDatasourcePropertyOperation);
+            }
+
+            operation = new ModelNode();
+            operation.get(OP).set("enable");
+            operation.get(OP_ADDR).set(address);
+
+            executeOperation(operation);
+
+            return address;
+        }
+    }
+    
+    protected void modifyModuleIronJacamarJdbc() throws Exception{
+    	addModule(moduleDefaultPath);
+    	createSubModuleJarConnectionListenerImpl(JAR_NAME);
+    	
+    	//we have to edit module.xml and add our new JAR as resource
```

Comment:
If this is the case, IMHO the feature is broken.

Plugin points exposed via classname attributes in the WildFly management model should also include an attribute that allows the user to specify the name of the module in which the class resides, and the subsystem should load the class from that module.

If a user alters one of our modules, a patch affecting that module will not apply.